node/src/chain/fsm.rs

```diff
@@ -505,168 +506,163 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network> InSyncImpl<DB, VM, N> {
+            // Ensure remote_blk is higher than the last finalized
             if remote_height
                 <= acc.get_latest_final_block().await?.header().height
             {
                 return Ok(None);
             }
 
-            // If our local chain has a block L_B with ConsensusState not Final,
-            // and we receive a block R_B such that:
-            //
-            // R_B.PrevBlock == L_B.PrevBlock
-            // R_B.Iteration < L_B.Iteration
-            //
-            // Then we fallback to N_B.PrevBlock and accept N_B
-            let result = acc.db.read().await.view(|t| {
-                if let Some(prev_header) =
-                    t.fetch_block_header(&remote_blk.header().prev_block_hash)?
-                {
-                    let local_height = prev_header.height + 1;
-                    if let Some(l_b) = t.fetch_block_by_height(local_height)? {
-                        if remote_blk.header().iteration
-                            < l_b.header().iteration
-                        {
-                            return Ok(Some((
-                                l_b.header().clone(),
-                                prev_header.state_hash,
-                            )));
-                        }
-                    }
+            // Ensure the block is different from what we have in our chain
+            if remote_height == tip_header.height {
+                if remote_header.hash == tip_header.hash {
+                    return Ok(None);
                 }
-
-                anyhow::Ok(None)
-            })?;
-
-            if let Some((local_header, state_hash)) = result {
-                match fallback::WithContext::new(acc.deref())
-                    .try_revert(
-                        &local_header,
-                        remote_blk.header(),
-                        RevertTarget::Commit(state_hash),
-                    )
+            } else {
+                let blk_exists = acc
+                    .db
+                    .read()
                     .await
-                {
-                    Ok(_) => {
-                        counter!("dusk_fallback_count").increment(1);
-                        if remote_height == acc.get_curr_height().await + 1 {
-                            acc.try_accept_block(remote_blk, true).await?;
-                            return Ok(None);
-                        }
-                    }
-                    Err(e) => {
-                        error!(
-                            event = "fallback failed",
-                            height = local_header.height,
-                            remote_height,
-                            err = format!("{:?}", e)
-                        );
-                        return Ok(None);
-                    }
+                    .view(|t| t.get_block_exists(&remote_header.hash))?;
+
+                if blk_exists {
+                    return Ok(None);
```

Comment:
Pls, move this check before the "get_last_final_block()` one

Reply:
Done